FirebaseAdmin/FirebaseAdmin.IntegrationTests/FirebaseAuthTest.cs

```diff
@@ -218,13 +220,49 @@ public async Task UserLifecycle()
                 Assert.False(user.Disabled);
                 Assert.NotNull(user.UserMetaData.CreationTimestamp);
                 Assert.Null(user.UserMetaData.LastSignInTimestamp);
-                Assert.Equal(2, user.ProviderData.Length);
+                Assert.Equal(3, user.ProviderData.Length);
+                var providerIds = new HashSet<string>();
+                foreach (ProviderUserInfo providerData in user.ProviderData)
+                {
+                    providerIds.Add(providerData.ProviderId);
+                }
+
+                Assert.Equal(providerIds, new HashSet<string>() { "phone", "password", "google.com" });
                 Assert.Empty(user.CustomClaims);
 
                 // Get user by email
                 user = await FirebaseAuth.DefaultInstance.GetUserByEmailAsync(randomUser.Email);
                 Assert.Equal(uid, user.Uid);
 
+                // Delete the linked provider and phone number
+                var unlinkArgs = new UserRecordArgs()
```

Comment:
Shall we merge this into the existing logic that disables the user account and removes other properties? 

```
               var disableArgs = new UserRecordArgs()
                {
                    Uid = uid,
                    Disabled = true,
                    DisplayName = null,
                    PhoneNumber = null,
                    PhotoUrl = null,
                    ProvidersToDelete = ....
                };
```